repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/headertranslation/HeaderTranslationWQualityTest.groovy

```diff
@@ -53,6 +52,42 @@ class HeaderTranslationWQualityTest extends ReposeValveTest {
         }
     }
 
+    def "When translate header with quality" () {
+
+        when: "Send request with headers to translate"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "GET", headers:
+                ["x-pp-user": "a", "x-tenant-name": "b", "x-roles": "c"])
+        def reposehandling = mc.getHandlings()[0]
+
+        then: "new header should include quality"
+        reposehandling.request.headers.contains("x-pp-user")
+        reposehandling.request.headers.contains("x-rax-username")
+        !reposehandling.request.headers.contains("x-tenant-name")
+        reposehandling.request.headers.contains("x-roles")
+        reposehandling.request.headers.contains("x-rax-roles")
+        reposehandling.request.headers.getFirstValue("x-rax-username") == "a;q=0.5"
+        reposehandling.request.headers.getFirstValue("x-rax-roles") == "c;q=0.2"
+        reposehandling.request.headers.getFirstValue("x-roles") == "c"
+        reposehandling.request.headers.getFirstValue("x-pp-user") == "a"
+    }
+
+    def "the original header is splittable, all new translated headers will be added with quality" () {
+        when: "client passes a request through repose with headers to be translated"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "GET",
+                headers: ["x-pp-user"    : "test, repose",
+                          "x-tenant-name": "b",
+                          "x-roles"      : "test"]
+        )
+        def reposehandling = mc.getHandlings()[0]
+
+        then:
+        reposehandling.request.getHeaders().contains("x-rax-username")
+        reposehandling.request.getHeaders().findAll("x-rax-username").contains("test;q=0.5")
+        reposehandling.request.getHeaders().findAll("x-rax-username").contains("repose;q=0.5")
```

Comment:
She updated the name of that test and added the tests I described in her latest commits.